src/test/java/org/broadinstitute/hellbender/tools/HtsgetReaderIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+package org.broadinstitute.hellbender.tools;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.testutils.ArgumentsBuilder;
+import org.broadinstitute.hellbender.testutils.IntegrationTestSpec;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+
+public class HtsgetReaderIntegrationTest extends CommandLineProgramTest {
```

Comment:
These tests are good but are probably only downloading single chunks.  I think a lot of that is out of our control unless we can control the server or want to download really large files.  

Have you run large tests on full size files including in parallel?  Could you create some tests on large files that exercise downloading multiple blocks, and check them in disabled so people can run them manually if they want?  

It would also be good to have tests for the response directly that show we are seeing both dataURI and httpUri responses.  Or if we're not, just clarifying which we're getting.